Sources/SwiftLanguageService/SignatureHelp.swift

```diff
@@ -36,16 +36,23 @@ fileprivate extension String {
 }
 
 fileprivate extension ParameterInformation {
-  init?(_ parameter: SKDResponseDictionary, _ signatureLabel: String, _ keys: sourcekitd_api_keys) {
+  init?(
+    _ parameter: SKDResponseDictionary,
+    _ signatureLabel: String,
+    _ keys: sourcekitd_api_keys,
+    _ parameterDocumentation: [String: String]
+  ) {
     guard let nameOffset = parameter[keys.nameOffset] as Int?,
       let nameLength = parameter[keys.nameLength] as Int?
     else {
       return nil
     }
 
     let documentation: StringOrMarkupContent? =
-      if let docComment: String = parameter[keys.docComment] {
-        .markupContent(MarkupContent(kind: .markdown, value: docComment))
+      if let name = parameter[keys.name] as String?,
+        let documentation = parameterDocumentation[name]
+      {
+        .markupContent(MarkupContent(kind: .markdown, value: documentation))
```

Comment:
Should we fall back to the entire doc comment if we can’t extract the parameter comment for some reason?

Reply:
Do you mean `parameter[keys.docComment]` or do you mean actually returning the entire signature documentation (without separating parameters) when we don't find the signature documentation one of the parameters?

For `parameter[keys.docComment]`, sourcekitd doesn't currently return anything there so it shouldn't make a difference at the moment and if we ever change sourcekitd to actually return a doc comment for parameters we should probably also update SourceKit-LSP to use that instead of having its own parameter extraction logic.

For option 2, I don't think that would be consistent with existing implementation of signature help, some parameters may not need specific explanation and thus we shouldn't show other parameters' documentation instead here.

WDYT?